### PR TITLE
AK: Make some tweaks to AK::Span.

### PR DESCRIPTION
**Pull request by asynts**
This pull request contains a few minor tweaks,

  - renames the `span()` methods on containers to `bytes()` if it is known at compile time that bytes are contained, and

  - adds some utility functions like `Span::copy_to()` or `Span::fill()` that are abstracting for `memcpy()`, `memmove()` and `memset()`.

### CODE REVIEW

File: AK/Span.h [hunk 1]
```diff
@@ -150,12 +148,47 @@ class Span : public Detail::Span<T> {
...
+        ASSERT(other.size() >= size());
+        __builtin_memcpy(other.data(), data(), sizeof(T) * size());
+    }
+
+    ALWAYS_INLINE void copy_trimmed_to(Span other) const
+    {
+        __builtin_memcpy(other.data(), data(), sizeof(T) * min(size(), other.size()));
+    }
+
+    ALWAYS_INLINE void move_to(Span other) const
```
[the remark this answers is not in the document]
Comment (asynts) [reply AK/Span.h:174]: That's a really good point, the default should be to assume overlapping spans. But `copy_to_nonoverlapping` is quite a mouthful and the common case is that the memory regions are not overlapping, I mean how many times do you use `memmove`?

Also `copy_trimmed_to` would become something like `copy_trimmed_nonoverlapping_to` that's really to much.

Maybe we can add an enum?

~~~c++
enum class Overlapping { Yes, No };

void copy_to(Span, Overlapping = Overlapping::Yes);
~~~

I can setup a pull request for that.

[the remark this answers is not in the document]
Comment (nico) [reply AK/Span.h:174]: That's a nice API, but I'd personally wait with adding it until there's a clear need. I'd just always call memmove() and only add the explicit nonoverlapping bit once there's a proven need. But if you feel having both is important, I think your proposal is better than what's in the tree :)

File: Libraries/LibCrypto/Cipher/Cipher.h [hunk 2]
```diff
@@ -64,8 +64,8 @@ struct CipherBlock {
     virtual ByteBuffer get() const = 0;
     virtual const ByteBuffer& data() const = 0;
 
-    virtual void overwrite(const ReadonlyBytes&) = 0;
-    virtual void overwrite(const ByteBuffer& buffer) { overwrite(buffer.bytes()); }
```
Comment (nico) [inline Libraries/LibCrypto/Cipher/Cipher.h]: clang points out:

```
../Libraries/LibCrypto/Cipher/Cipher.h:68:54: error: all paths through this function will call itself [-Werror,-Winfinite-recursion]
    virtual void overwrite(const ByteBuffer& buffer) { overwrite(buffer); }
                                                     ^
```

File: Libraries/LibCrypto/Cipher/AES.h [hunk 3]
```diff
@@ -54,8 +54,8 @@ struct AESCipherBlock : public CipherBlock {
     virtual ByteBuffer get() const override { return m_data; };
     virtual const ByteBuffer& data() const override { return m_data; };
 
-    virtual void overwrite(const ReadonlyBytes&) override;
-    virtual void overwrite(const ByteBuffer& buffer) override { overwrite(buffer.bytes()); }
+    virtual void overwrite(ReadonlyBytes) override;
+    virtual void overwrite(const ByteBuffer& buffer) override { overwrite(buffer); }
```
Comment (nico) [inline Libraries/LibCrypto/Cipher/AES.h:58]: clang points out:

```
../Libraries/LibCrypto/Cipher/AES.h:58:63: error: all paths through this function will call itself [-Werror,-Winfinite-recursion]
    virtual void overwrite(const ByteBuffer& buffer) override { overwrite(buffer); }
                                                              ^
```

File: AK/Span.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (asynts) [reply AK/Span.h:174]: > That's a nice API, but I'd personally wait with adding it until there's a clear need.

I agree. #3179

File: Libraries/LibCrypto/Cipher/Cipher.h (hunk 2 above)
Comment (asynts) [reply Libraries/LibCrypto/Cipher/Cipher.h]: Oops. #3179.

File: Libraries/LibCrypto/Cipher/AES.h (hunk 3 above)
Comment (asynts) [reply Libraries/LibCrypto/Cipher/AES.h:58]: Oops. #3179.

**Pull request merged**